cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala

```diff
@@ -116,6 +118,39 @@ class PredicateFunctionsTest extends DocumentingTest {
         resultTable()
       }
     }
+    section("isEmpty()", "functions-isEmpty") {
+      p("`isEmpty()` returns true if the given list or map contains no elements or if the given string contains no characters.")
+      function("isEmpty(list)", "A Boolean.", ("list", "An expression that returns a list."))
+      query(
+        """MATCH (n)
+          |WHERE NOT isEmpty(n.array)
+          |RETURN n.name AS name""".stripMargin, ResultAssertions(r => {
+          r.toList should equal(List(Map("name" -> "Alice"), Map("name" -> "Bob"), Map("name" -> "Charlie"), Map("name" -> "Daniel"), Map("name" -> "Eskil"), Map("name" -> null)))
+        })) {
+        p("The names of all nodes with the `array` property being non-empty are returned.")
+        resultTable()
+      }
+      function("isEmpty(map)", "A Boolean.", ("map", "An expression that returns a map."))
+      query(
+        """MATCH (n)
+          |WHERE NOT isEmpty(properties(n))
+          |RETURN n.name AS name""".stripMargin, ResultAssertions(r => {
+          r.toList should equal(List(Map("name" -> "Eskil")))
+        })) {
+        p("The names of all nodes with that have at least one property are returned. Not that this includes nodes not having the `name` property.")
+        resultTable()
+      }
+      function("isEmpty(string)", "A Boolean.", ("string", "An expression that returns a string."))
+      query(
+        """MATCH (n)
+          |WHERE isEmpty(n.eyes)
+          |RETURN n.age AS age""".stripMargin, ResultAssertions(r => {
+          r.toList should equal(List(Map("age" -> 61)))
+        })) {
+        p("The ages of all nodes with that have the empty string as the `eyes` property.")
```

Comment:
It only occurred to me now, that `WHERE isEmpty(n.eyes)` will keep rows where `n.eyes = ''` but drop rows where `n.eyes = null` (i.e. unset). That might seem surprising to a user, so perhaps it should be called out here somewhere?